src/chat-bubble/__tests__/chat-bubble.test.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { cleanup, render } from "@testing-library/react";
+import { afterEach, describe, expect, test } from "vitest";
+
+import ButtonGroup from "@cloudscape-design/components/button-group";
+import ExpandableSection from "@cloudscape-design/components/expandable-section";
+
+import { Avatar } from "../../../lib/components";
+import ChatBubble, { ChatBubbleProps } from "../../../lib/components/chat-bubble";
+import createWrapper from "../../../lib/components/test-utils/dom";
+
+import styles from "../../../lib/components/chat-bubble/styles.selectors.js";
+
+function renderChatBubble(props: ChatBubbleProps) {
+  const { container } = render(<ChatBubble {...props} />);
+
+  return createWrapper(container).findChatBubble()!;
+}
+
+describe("Chat bubble", () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  test("Can access slots and elements inside", () => {
+    const wrapper = renderChatBubble({
+      type: "sent",
+      avatar: <Avatar ariaLabel="Avatar" />,
+      children: (
+        <>
+          Content
+          <ExpandableSection headerText="Sources">Sources</ExpandableSection>
+        </>
+      ),
+      ariaLabel: "Chat bubble",
+      actions: (
+        <ButtonGroup
+          variant="icon"
+          items={[
+            {
+              type: "group",
+              text: "Feedback",
+              items: [
+                {
+                  type: "icon-button",
+                  id: "helpful",
+                  iconName: "thumbs-up",
+                  text: "Helpful",
+                },
+                {
+                  type: "icon-button",
+                  id: "not-helpful",
+                  iconName: "thumbs-down",
+                  text: "Not helpful",
+                },
+              ],
+            },
+          ]}
+        />
+      ),
+    });
+
+    expect(wrapper.findAvatarSlot()?.findAvatar()?.getElement()).toBeVisible();
```

Comment:
It is better to use non-null assertion with `!` instead of optional chaining `?` in the tests. There is no difference for this test, but when asserting against non-visible, with the optional chaining you won't be able to tell was it the avatar or the avatar slot missing.